HW0/P4/P4.py

```diff
@@ -30,47 +30,106 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    s_true = np.loadtxt('P4_trajectory.txt', delimiter=',')
+    ax.plot(s_true[:,0], s_true[:,1], s_true[:,2],
+            '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    m = np.loadtxt('P4_measurements.txt', delimiter=',')
+    s_obs = m * np.array([1./rx, 1./ry, 1./rz]) 
+    ax.plot(s_obs[:,0], s_obs[:,1], s_obs[:,2],
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    # Construct matrix A from blocks
+    sub_a = np.eye(3)
+    sub_b = np.eye(3) * dt
+    sub_c = np.zeros((3,3))
+    sub_d = np.eye(3) * (1-c*dt)
+    A = np.bmat([[sub_a, sub_b], [sub_c, sub_d]])
+    A = np.array(A)
```

Comment:
This is trivial, but the np.diag() function that you use below could be a way of achieving the same result with fewer lines of code.